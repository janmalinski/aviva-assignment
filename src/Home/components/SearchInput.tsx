import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

interface ISearchInputProps {
    text?: string;
    handleSetText: (text: string) => void;
    navigateToWeatherDetails: () => void;
    testID?: string
};

const SearchInput = ({text, handleSetText, navigateToWeatherDetails, testID}: ISearchInputProps) => {
  return (
    <View testID={testID} style={[styles.row, styles.searchContainer]}>
        <TextInput testID='search-text-input' value={text} style={styles.textInput} placeholder='Search' onChangeText={text => handleSetText(text)} />
        <Ionicons testID='search-button' style={styles.searchButton} name='search-outline' size={30} onPress={navigateToWeatherDetails} />
    </View>
  )
};

export default React.memo(SearchInput);

const styles = StyleSheet.create({
    searchContainer: {
        flexDirection: 'row',
        height: 40,
        marginTop: 30,
        marginBottom: 15
    },
    textInput:  {
        flex: 1,
        borderBottomWidth: 1,
        height: 40,
        paddingHorizontal: 5,
      },
    searchButton: {
    alignSelf: 'center'
    },
    row: {
        flexDirection: 'row',
    },
})