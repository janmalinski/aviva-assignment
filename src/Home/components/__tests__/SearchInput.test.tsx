import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';


import SearchInput from '../SearchInput';

const text = 'PoznaÅ„';
const handleSetText = jest.fn();
const navigateToWeatherDetails = jest.fn();


describe('Rendering SearchInput', () => {
    const SearchInputEl = (
        <SearchInput testID='search-input' text={text} handleSetText={handleSetText} navigateToWeatherDetails={navigateToWeatherDetails} />
    );

    it('Should match a snapshot', () => {
        const elJson = render(SearchInputEl).toJSON();
        expect(elJson).toMatchSnapshot(); 
    });
});

describe('Behaviour SearchInput', () => {
    const searchInputEl = (<SearchInput testID='search-input' text={text} handleSetText={handleSetText} navigateToWeatherDetails={navigateToWeatherDetails} />);
    it('Should apply the text prop value', () => {
        const { getByTestId } = render(searchInputEl);
        const textInput = getByTestId('search-text-input');
        expect(textInput).toBeTruthy();
        expect(textInput.props.value).toBe(text);
    });

    it('Should press search button', () => {
        const { getByTestId } = render(searchInputEl);
        const searchButton = getByTestId('search-button');
        expect(searchButton).toBeTruthy();
        fireEvent.press(searchButton);
    })


});
