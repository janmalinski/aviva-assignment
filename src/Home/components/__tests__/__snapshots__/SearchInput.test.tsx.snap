// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering SearchInput Should match a snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 15,
        "marginTop": 30,
      },
    ]
  }
  testID="search-input"
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search"
    style={
      {
        "borderBottomWidth": 1,
        "flex": 1,
        "height": 40,
        "paddingHorizontal": 5,
      }
    }
    testID="search-text-input"
    value="PoznaÅ„"
  />
  <Text />
</View>
`;
